x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRuleset.java

```diff
@@ -64,7 +55,7 @@ public QueryRuleset(StreamInput in) throws IOException {
         (params, resourceName) -> {
             final String id = (String) params[0];
             // Check that id matches the resource name. We don't want it to be updatable
-            if (id.equals(resourceName) == false) {
+            if (id != null && id.equals(resourceName) == false) {
```

Comment:
Note: This is the one check that we still do that will fail before license checks occur. The solution is to remove the id from the request - this will happen already as part of a different PR.

Reply:
We can probably move this validation to the request. The `validate()` method could check that the `resourceName` is the same as the `QueryRuleset` `id`, and fail otherwise. WDYT?